### ISSUE TITLE
How exactly are the predictions made for ignore_label handled in the panoptic deeplab model?

### ISSUE THREAD

**Issue by free-bit** [google-research/deeplab2#43]
While analyzing the files related to panoptic deeplab model for my research, I noticed that the `output_channels` variable of the segmentation head of the panoptic deeplab models is configured so that the resulting semantic map has dimensions equal to the `num_classes` of the respective dataset, which includes the `ignore_label`. From that, it seems like the model is actually computing a score for the `ignore_label` as well. 

To make it more concrete, if we take the panoptic deeplab model with the COCO panoptic dataset as an example, there are overall 134 classes in this dataset, including the 'unknown' class with `ignore_label=0`, and the semantic head of the model is configured to give 134-channel output in the proto files.

I understand how the loss is configured to ignore the 'unknown' class during training but I couldn't find where the label prediction for the `ignore_label` is explicitly avoided. Again, in the example of panoptic deeplab, as far as I see, the model has a postprocessing stage that converts the logits to semantic label predictions by taking argmax on the logits as usual. However, I checked the dimensions of this logits vector and it also has `num_channels` dimensions. Based on this observation, it seems to me like it is actually possible to predict the `ignore_label` but when I check the semantic predictions made by the model, I never observed such a thing. So, I believe the predictions for class 'unknown' is avoided somehow but I wonder how this is done. Is this happening during the training procedure implicitly and automatically (by simply learning not to assign high scores to this class)? Or, is there particular post-processing that you apply explicitly (e.g. if the label is `ignore_label`, take the second-highest value from the argmax)?

**Comment by csrhddlam** [comment]
Hi,

Thanks for raising the issue and checking all the details. As far as I know, there is no explicit `ignore_label` handling in the panoptic-deeplab post-processing. I think in the current implementation, the training "implicitly and automatically" learns not to predict `ignore_label`, since there is not any positive label for the `ignore_label` class. That said, in a sufficiently trained model, the current implementation will not be an issue.

If you want to be really rigorous, a simple diagnose could be done by setting a large positive or negative bias towards the `ignore_label` class in a trained (downloaded) model and checking the visualizations and results. But again, I don't think explicitly dealing with the `ignore_label` will result in any performance difference.

**Comment by free-bit** [comment]
Thank you very much for the explanation. It is very good to learn these details for me.
